src/libAtomVM/opcodesswitch.h

```diff
@@ -3338,27 +3344,22 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
                 #endif
 
                 #ifdef IMPL_EXECUTE_LOOP
-                    VERIFY_IS_INTEGER(src,  "bs_put_integer");
+                    VERIFY_IS_ANY_INTEGER(src, "bs_put_integer");
                     VERIFY_IS_INTEGER(size, "bs_put_integer");
                     VERIFY_IS_INTEGER(flags, "bs_put_integer");
 
-                    avm_int_t src_value = term_to_int(src);
+                    avm_int_t src_value = term_maybe_unbox_int(src);
```

Comment:
I think this will fail when we use it on an in64 integer on a 32 bit architecture. Did we test this scenario?